aws-operator-user/iam-policy.json

```diff
@@ -44,17 +40,13 @@
                 "iam:RemoveRoleFromInstanceProfile",
                 "iam:UpdateAssumeRolePolicy",
                 "iam:UpdateRoleDescription",
-                "kms:*",
```

Comment:
How does the operator work if it has to deal with KMS and S3 if we drop the permissions?

Reply:
IIRC we do deal with KMS and S3 in workload cluster accounts, the GiantSwarmUserAWSOperator policy is in the management cluster accounts.

Reply:
I see there is `aws-operator-user` and `aws-operator-role`. What is the difference? The readme just made me more bamboozled. 😄

Reply:
we have a user `GiantSwarmAWSOperator` which has policy `GiantSwarmUserAWSOperator` in management cluster AWS accounts. this policy is only used for permissions inside the management cluster AWS accounts.

additionally we have in each workload cluster AWS accounts a role `GiantSwarmAWSOperator` which has a policy `GiantSwarmAWSOperator`. The role has a trust identity to the `GiantSwarmAWSOperator` user. So whenever aws-operator needs to do something inside the workload cluster account it uses this policy.

Reply:
> we have a user `GiantSwarmAWSOperator` which has policy `GiantSwarmUserAWSOperator` in management cluster AWS accounts. this policy is only used for permissions inside the management cluster AWS accounts.

Since [9.1.2](https://github.com/giantswarm/aws-operator/blob/master/CHANGELOG.md#912---2020-10-15) aws-operator is accessing management cluster using role assumption. Eventually the only permission for this user should be `sts:AssumeRole`.

For the time being (i.e. as long as we still run 9.1.2) I'd simply link the same policy we use for the role https://github.com/giantswarm/giantswarm-aws-account-prerequisites/blob/39ea4d58d3593df376f0ddce1cd90b72bc3a1e00/aws-operator-role/iam-policy.json. Security-wise it doesn't really change anything because this user has a permission to assume this role.

Reply:
I guess, we do not roll it in existing installations, only when creating a new ones. Shall I drop the rest and only leave `"sts:AssumeRole"`?

Because when we would link the same policy we do not use s3 for instance in management cluster accounts but in workload cluster accounts, same for kms.

Reply:
i would say that `"trustedadvisor:*"` is somehow accessed when we check limits for CP accounts but might be wrong

 only a few customers have this enabled so e don't check it everywhere

Reply:
Good point, I reverted "trustedadvisor:*". It does not harm IMHO.